features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -161,13 +168,36 @@ internal class SessionReplayFeature(
 
     // endregion
 
+    // region Manual Recording
+
+    internal fun manuallyStopRecording() {
+        if (shouldRecord.get()) {
+            recordingStateChanged.set(true)
+            shouldRecord.set(false)
```

Comment:
Should `shouldRecord` be changed before `recordingStateChanged`?

Reply:
you should use `if(shouldRecord.compareAndSet(true, false))`